File: 2026-2/sesion-de-clase-5/dotnet/Lab01Lib/SingleChoiceQuestion.cs
namespace PUCP.Labs.Lab01.Preguntas;

using PUCP.Labs.Lab01.Examenes;

public class SingleChoiceQuestion : Question
{
    public int Respuesta { get; }

    public SingleChoiceQuestion(
            int codigo,
            string prompt,
            List<string> opciones,
            int respuesta)
        : base(codigo, prompt, opciones)
    {
        Respuesta = respuesta;
    }

    public override void DevolverDatos()
    {
        base.DevolverDatos();
        Console.WriteLine("Ingrese su respuesta: ");
    }

    public override void RecuperarRespuesta(AssessmentItem item)
    {
        int respuestaAlumno = int.Parse(Console.ReadLine()!);
        item.EsCorrecto = Respuesta == respuestaAlumno;
    }
}

File: 2026-2/sesion-de-clase-04/java/Lab01-2026-1/pe/edu/pucp/lab/preguntas/SingleChoiceQuestion.java
package pe.edu.pucp.lab.preguntas;

import java.util.List;
import java.util.Scanner;

import pe.edu.pucp.lab.examenes.AssessmentItem;

public class SingleChoiceQuestion extends Question {

    private int respuesta;

    public SingleChoiceQuestion(
            int codigo,
            String prompt,
            List<String> opciones,
            int respuesta) {
        super(codigo, prompt, opciones);
        this.respuesta = respuesta;
    }

    public int getRespuesta() {
        return respuesta;
    }

    @Override
    public void devolverDatos() {
        super.devolverDatos();
        System.out.println("Ingrese su respuesta: ");
    }

    @Override
    public void recuperarRespuesta(AssessmentItem item) {
        Scanner scanner = new Scanner(System.in);
        int respuestaAlumno = scanner.nextInt();
        item.setEsCorrecto(respuesta == respuestaAlumno);
    }
}
